### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by isinyaaa**
Some improvement suggestions from a quick look around. Tested with existing tests, but I still haven't been able to run e2e.

### CODE REVIEW

File: omlmd/cli.py [hunk 1]
```diff
@@ -26,9 +29,15 @@ def cli():
 @cli.command()
 @plain_http
 @click.argument("target", required=True)
-@click.option("-o", "--output", default=".", show_default=True)
+@click.option(
+    "-o",
+    "--output",
+    default=Path.cwd(),
+    show_default=True,
+    type=click.Path(path_type=Path, exists=True, resolve_path=True),
```
Comment (tarilabs) [inline omlmd/cli.py:37]: I believe here lies an issue as highlighted by E2E, the output path was previously created if unexisting

**Review by tarilabs — COMMENTED** [review]
thanks @isinyaaa much appreciated !

I'm not a big fan of reformattings, but I'd take it.

Can you kindly clarify/help me better understand the followings:

- [simplify type annotations](https://github.com/tarilabs/omlmd/pull/9/commits/1c7f52f78a4c96e6b55c9bc22d494c4653ec3872) [1c7f52f](https://github.com/tarilabs/omlmd/pull/9/commits/1c7f52f78a4c96e6b55c9bc22d494c4653ec3872)
to me that is python 3.10 and up: https://peps.python.org/pep-0604
btw, not sure why mypy is not complaining about this 🤔 

- [provider: skip mode-setting before download_blob](https://github.com/tarilabs/omlmd/pull/9/commits/b95b014aabc9caa06e51aa4002b115037cd910fd) [b95b014](https://github.com/tarilabs/omlmd/pull/9/commits/b95b014aabc9caa06e51aa4002b115037cd910fd)
why would I skip mode=w here for file which I want to write, please? 🤔 asking for improving my py knowledge! :D 

About the E2E test failure, see below:

**Comment by isinyaaa** [comment]
> thanks @isinyaaa much appreciated !
> 
> I'm not a big fan of reformattings, but I'd take it.

I can disable my auto-formatter if you prefer, but having some sort of style guidelines would be helpful for contributions.
 
> Can you kindly clarify/help me better understand the followings:
> 
> * [simplify type annotations](https://github.com/tarilabs/omlmd/pull/9/commits/1c7f52f78a4c96e6b55c9bc22d494c4653ec3872) [1c7f52f](https://github.com/tarilabs/omlmd/pull/9/commits/1c7f52f78a4c96e6b55c9bc22d494c4653ec3872)
>   to me that is python 3.10 and up: https://peps.python.org/pep-0604
>   btw, not sure why mypy is not complaining about this 🤔

when you declare `from __future__ import annotations` it enables forward compatibility for type annotations.

> * [provider: skip mode-setting before download_blob](https://github.com/tarilabs/omlmd/pull/9/commits/b95b014aabc9caa06e51aa4002b115037cd910fd) [b95b014](https://github.com/tarilabs/omlmd/pull/9/commits/b95b014aabc9caa06e51aa4002b115037cd910fd)
>   why would I skip mode=w here for file which I want to write, please? 🤔 asking for improving my py knowledge! :D

You don't need it because `oras` is already opening that file in the required mode (`w+b`, not `w`).

**Comment by tarilabs** [comment]
thanks for the clarification @isinyaaa , can you please reword your commits to sign-off for DCO, ...or just squash them if easier to add the required DCO

**Comment by tarilabs** [comment]
many thanks again, proceeding to merge

**Pull request merged**